test/extended/util/framework.go

```diff
@@ -56,18 +60,99 @@ func WaitForInternalRegistryHostname(oc *CLI) (string, error) {
 		imageConfig, err := oc.AsAdmin().AdminConfigClient().ConfigV1().Images().Get("cluster", metav1.GetOptions{})
 		if err != nil {
 			if errors.IsNotFound(err) {
+				e2e.Logf("Image config object not found")
 				return false, nil
 			}
+			e2e.Logf("Error accessing image config object: %#v", err)
 			return false, err
 		}
 		if imageConfig == nil {
+			e2e.Logf("Image config object nil")
 			return false, nil
 		}
 		registryHostname = imageConfig.Status.InternalRegistryHostname
 		if len(registryHostname) == 0 {
+			e2e.Logf("Internal Registry Hostname is not set in image config object")
 			return false, nil
 		}
-		return true, nil
+
+		// now confirm the OCM was restarted after the image registry was set;
+		// to do that, we are going to compare the create time of the ocm pods
+		// with the last transition times of the ocm operator object progressing condition;
+		// we'll also confirm timings between the OCM rollout and image config creation,
+		// and we'll verify that the OCM config's internal registry hostname matches
+		// the image config's internal registry hostname
+		// get the OCM cfg
```

Comment:
does the controller manager log anything when it starts up, about what it knows the internal registry hostname to be?  perhaps checking the logs of the controller manager pod could be an easier/more reliable way to check this?  (i'd even be ok adding such a log output to the controller manager if necessary... it's probably something we want to know anyway)

Reply:
This does show up in the logs:

```
I0813 15:21:47.865941       1 docker_registry_service.go:284] Updating registry URLs from map[172.30.122.228:5000:{} image-registry.openshift-image-registry.svc.cluster.local:5000:{} image-registry.openshift-image-registry.svc:5000:{}] to map[172.30.122.228:5000:{} image-registry.openshift-image-registry.svc.cluster.local:5000:{} image-registry.openshift-image-registry.svc:5000:{}]
I0813 15:21:47.866011       1 deleted_token_secrets.go:72] caches synced
I0813 15:21:47.899404       1 build_controller.go:466] Starting build controller
```

that said, though less code for sure, parsing the pod logs seems more fragile, no?

Reply:
We can make it not fragile by introducing a test that ensures that log message is always output (so that we do not accidentally remove the log output in the future).

Reply:
(in fairness i have not looked through your current implementation of the check based on timestamps/etc, so let me do that and give you a more informed opinion)

Reply:
dumping those pod logs if this logic finds a problem, or may dump the pod  logs and filter for build_controller.go and docker_registry_service.go regardless, would be useful info if the hiccup occurred again

So maybe the approach here *plus* that?